dataprofiler/profilers/profile_builder.py

```diff
@@ -80,16 +81,31 @@ def __init__(self, df_series, sample_size=None, min_sample_size=5000,
             'data_stats_profile':
                 ColumnStatsProfileCompiler(clean_sampled_df, self.options)}
 
-        # use the data labeler by default
-        use_data_labeler = True
-        if options and isinstance(options, StructuredOptions):
-            use_data_labeler = options.data_labeler.is_enabled
-
-        if use_data_labeler:
+        data_labeler = self.options.data_labeler.data_labeler_model
+        if data_labeler:
             self.profiles.update(
                 {'data_label_profile':
                      ColumnDataLabelerCompiler(clean_sampled_df, self.options)})
 
+        else:
+            # use the data labeler by default
+            use_data_labeler = True
+            if options and isinstance(options, StructuredOptions):
+                use_data_labeler = options.data_labeler.is_enabled
+
+            if use_data_labeler:
+                data_labeler_dirpath = None
+                if options and options.data_labeler_dirpath:
+                    data_labeler_dirpath = options.data_labeler_dirpath
+
+                self.options.data_labeler.data_labeler_model = \
+                    DataLabeler(labeler_type='structured',
+                                dirpath=data_labeler_dirpath,
+                                load_options=None)
```

Comment:
Isn't this effectively the same code as the column: `data_labeler_column_profile.py`